doc/build.txt

```diff
@@ -21,47 +21,80 @@ CONTENTS                                                         *build-contents
 belongs to a project with a build system or not. Thus it knows how to build
 and run it. This plugin works by searching from the files directory upwards,
 until it finds a known build file.
+It tries to provide a homogeneous interface for common tasks, independently of
+the underlying build system (or its absence).
 
 ===============================================================================
 2. Commands                                                      *build-commands*
 
--------------------------------------------------------------------------------
-2.1 Build a specific target                                               *Build*
+This plugin propose few commands:
+- `:Build` a general purpose command used to achieve common tasks
+- `:BuildInit` initialize the build system if needed
+- `:BuildInfo` provide informations and help message on the current build
+  system
 
-Try to build the given optional target with the specified optional arguments.
-If the current file does not belong to a build system, it will be build using
-associated compilers.
+-------------------------------------------------------------------------------
+2.1 Common tasks                                                          *Build*
+
+The `:Build` command takes either no arguments or a first argument called a
+subcommand and any number of other arguments as you would pass them to a shell
+command.
+
+Subcommands should represent the same action across different build systems
+independently from how each works.
+
+For example `:Build clean` on a `Make` based project would expand to 
+`make clean` while for a `CMake` based project it would result in
+`rm -r build/`
```

Comment:
Don't forget to change this CMake command example.

Reply:
What would you expect ?

Reply:
`cmake --build ./build/ -- clean`

Reply:
This is consistent with `s:build_systems`. Almost every project has a `clean` target.